data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -179,22 +188,61 @@ public void doOutput(final Collection<Record<Event>> records) {
 
   }
 
-  private Optional<String> getDocumentIdFromDocument(final SerializedJson document) {
+  /*
+   * Look for 'field' in the 'map' and if present, return it's value.
+   * The field may contain multiple sub-fields separated by "/", in which
+   * case, each subfield (except the last one) should be in a map inside 
+   * the map and it is searched for next subfield.
+   */
+  private String findFieldValueInMap(String field, Map map) {
```

Comment:
The `Event` model already has support for reading fields. 

I'd prefer to see a solution where `Event::get` is used directly. Then we don't need this code at all. We are moving away from supporting non Event models. We don't need to support the nested syntax for Maps that are not Events.

Please see the following method.

https://github.com/opensearch-project/data-prepper/blob/8977667a925691f1f8584ff04fbf739cfa26689b/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java#L198-L200

It may make sense to have this `SerializedJson` retrieve the docId and routing value and then you can use it later.

Reply:
OK. Makes sense. Will look into SerializedJson